source/test/TestApp/PlayFabEventTest.cpp

```diff
@@ -222,19 +244,19 @@ namespace PlayFabUnit
         std::atomic<uint32_t> eventsRemaining(pNumThreads * pNumEventsPerThread);
         for (uint32_t thread = 0; thread < pNumThreads; ++thread)
         {
+            std::shared_ptr<PlayFabEventAPI> api = SetupEventTest();
             testThreadPool.emplace_back(
-                [&eventsRemaining, pNumEventsPerThread, this]()
+                [api, &eventsRemaining, pNumEventsPerThread, this]()
             {
                 try
                 {
-                    std::shared_ptr<PlayFabEventAPI> api = SetupEventTest();
                     for (uint32_t i = 0; i < pNumEventsPerThread; ++i)
                     {
                         api->EmitEvent(MakeEvent(PlayFabEventType::Default),
                             [&eventsRemaining, this]
                         (std::shared_ptr<const PlayFab::IPlayFabEvent>, std::shared_ptr<const PlayFab::IPlayFabEmitEventResponse>)
                         {
-                            if (--eventsRemaining == 0)
+                            if (--eventsRemaining <= 0)
```

Comment:
this is alarming.  why would we get to negative events?

Reply:
So, I was trying to be proactive, and also guessing at possible causes for failure in the PlayStation.
It didn't pan out. This test passes on everything but PS, where it just crashes.
BUT, I think this is still a good because:
1. If Pass() gets called twice, it will result in a test failure (cant call pass twice)
2. If some future change or circumstance causes more than the expected events to occur, the test will start failing (can't call Pass() twice).
3. Without this change, if more than the expected number of events occur, they'll be ignored and the test will pass, because we only call Pass() when ==0, which will only be once.

---
In reply to: [356226877](https://github.com/PlayFab/XPlatCoreTemplate/pull/62#discussion_r356226877) [](ancestors = 356226877)

Reply:
Wouldn't it be a red flag if more than the expected number of events occur?

Reply:
Yes, that's exactly right. And now, and most importantly unlike before; the test will fail if that occurs.

Reply:
ah ok